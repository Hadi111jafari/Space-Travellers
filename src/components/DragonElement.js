import React from 'react';
import { useDispatch } from 'react-redux';
import PropTypes from 'prop-types';
import { reserveDragons, unreserveDragons } from '../redux/dragons/dragon';

const DragonElement = (props) => {
	const dispatch = useDispatch();
	const {
		id, name, dragon_type, flickrImages, reserved,
	} = props;

	DragonElement.defaultProps = {
		reserved: false,
	};

	const reserveDragonHandler = (e) => {
		if (reserved) {
			dispatch(unreserveDragons(e.target.id));
		} else {
			dispatch(reserveDragons(e.target.id));
		}
	};
	const reservation = reserved ? 'Cancel Reservation' : 'Reserve Dragon';
	const btnClass = reserved ? 'grayBtn' : 'blueBtn';
	return (
		<div className="DragonEl">
			<div className="dragonImg">
				<img src={flickrImages} alt={dragon_type} />
			</div>
			<div>
				<h2>{name}</h2>
				<p>
					{(reserved) && (<span className="badge">Reserved</span>)}
					{dragon_type}
				</p>
				<button className={btnClass} type="button" id={id} onClick={reserveDragonHandler}>{reservation}</button>
			</div>
		</div>
	);
};

DragonElement.propTypes = {
	id: PropTypes.string.isRequired,
	name: PropTypes.string.isRequired,
	reserved: PropTypes.bool,
	description: PropTypes.string.isRequired,
	flickrImages: PropTypes.string.isRequired,
};

export default DragonElement;
