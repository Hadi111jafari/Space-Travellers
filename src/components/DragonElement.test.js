import { Provider } from 'react-redux';
import renderer from 'react-test-renderer';
import store from '../redux/configureStore';
import DragonElement from './DragonElement.js';

const dragon = {
  dragon_id: '654',
  dragon_name: 'Dragon',
  description: 'Lorem',
  flickr_images: ['URL'],
  reserved: false,
};

describe('DragonElement component', () => {
  test('component match with the snapshot', () => {
    const tree = renderer.create(
      <Provider store={store}>
        <DragonElement
          key={dragon.dragon_id}
          id={dragon.dragon_id}
          name={dragon.dragon_name}
          description={dragon.description}
          flickrImages={dragon.flickr_images[0]}
          reserved={dragon.reserved}
        />
      </Provider>,
    ).toJSON();
    expect(tree).toMatchSnapshot();
  });
});
