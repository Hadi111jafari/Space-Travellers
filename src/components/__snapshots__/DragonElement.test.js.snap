// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`DragonElement component component match with the snapshot 1`] = `
<div
  className="dragonEl"
>
  <div
    className="dragonImg"
  >
    <img
      alt="Lorem"
      src="URL"
    />
  </div>
  <div>
    <h2>
      Dragon
    </h2>
    <p>
      Lorem
    </p>
    <button
      className="blueBtn"
      id="654"
      onClick={[Function]}
      type="button"
    >
      Reserve Dragon
    </button>
  </div>
</div>
`;
